public/javascript/create-issue.js

```diff
@@ -12,103 +12,100 @@ function clearCreateIssueForm() {
 
 // Create new issue handler
 async function createIssue(event) {
-  event.preventDefault();
+    event.preventDefault();
 
-  // Get form values to pass as object
-  const title = document.getElementById('create-issue-title').value;
-  const dueDate = document.getElementById('create-issue-duedate').value;
-  const priority = document.getElementById('create-issue-priority').value;
-  const label = document.getElementById('create-issue-labels').value;
-  const description = document.getElementById('create-issue-body').value;
-  const projectId = document.getElementById('project-dropdown').value;
+    // Get form values to pass as object
+    const title = document.getElementById('create-issue-title').value;
+    const dueDate = document.getElementById('create-issue-duedate').value;
+    const priority = document.getElementById('create-issue-priority').value;
+    const label = document.getElementById('create-issue-labels').value; 
+    const description = document.getElementById('create-issue-body').value;
+    const projectId = document.getElementById('project-dropdown').value;
 
-  // check if minimum values are provided
-  if (projectId && title && description && dueDate && priority && label) {
+    // check if minimum values are provided
+    if(projectId && title && description && dueDate && priority) {
 
-    let postBody = {
-      due_date: dueDate,
-      priority: priority,
-      issueState_id: 1,
-      project_id: projectId,
-      data: {
+      const data = {
         title: title,
         body: description,
-        state: 'open',
-        labels: [label]
+        state: 'open'
       }
-    };
 
-    // Post data to /api/project/ route
-    const response = await fetch(`/api/issue/`, {
-      method: 'POST',
-      body: JSON.stringify(postBody),
-      headers: {
-        'Content-Type': 'application/json'
+      if (label !== "no-label" && label) {
+        data.labels = [label];
       }
-    });
 
-    if (response.ok) {
-      console.log(response);
-
-      document.location.replace('/dashboard/')
+        // Post data to /api/project/ route
+        const response = await fetch(`/api/issue/`, {
+            method: 'POST',
+            body: JSON.stringify({
+              due_date: dueDate,
+              priority: priority,
+              issueState_id: 1,
+              project_id: projectId,
+              data: data
+            }),
+            headers: {
+              'Content-Type': 'application/json'
+            }
+          });
+        
+          if (response.ok) {
+            console.log(response);
+            document.location.replace('/dashboard/');
+          } else {
+            alert(response.statusText);
+        }
     } else {
-      alert(response.statusText);
+        alert('Missing information.');
     }
-
     // Close modal
```

Comment:
Move this back inside the if statement. If you leave this here and there is a problem the form closes and erases the data so you can't fix it.